community/kernel/src/main/java/org/neo4j/kernel/impl/api/index/IndexingService.java

```
@@ -433,13 +434,25 @@ private void apply( IndexUpdates updates, IndexUpdateMode updateMode )
         {
             for ( NodeUpdates update : updates )
             {
-                for ( LabelSchemaDescriptor descriptor : updaterMap.descriptors() )
+                for ( IndexEntryUpdate indexUpdate : update.forIndexes( updaterMap.descriptors(), ( nodeId, propertyId ) ->
                 {
-                    Optional<IndexEntryUpdate> entry = update.forIndex( descriptor );
-                    if ( entry.isPresent() )
+                    try
                     {
-                        updaterMap.getUpdater( descriptor ).process( entry.get() );
+                        return storeView.getProperty( nodeId, propertyId ).value();
                     }
+                    catch ( EntityNotFoundException e )
+                    {
+                        // TODO: we should not use exceptions for this, because that is slow
+                        return null;
+                    }
```

This catch block should never be reached in this case, so it's not so bad.

What might be worse than the slowness is that if this happens it would be a symptom of something gone quite horribly wrong, so perhaps we shouldn't just silently ignore it like this...